Document/0x06i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -491,10 +491,39 @@ However, these do not include all the libraries being used. Next, with Class-dum
 $ ./class-dump <Executable> -r
 ```
 
+### Memory Corruption Bugs
+iOS applications have various ways to run into memory corruption bugs: first there are the native code issues which have been mentioned in the general Memory Corruption Bugs section. Next, there are various unsafe operations with both Objective-c and Swift to actually wrap around native code which can create issues. Last, both Swift and Objective-C implementations can result in memory leaks due to retaining objects which are no longer in use.
+
+#### Static Analysis
+Are there native code parts? If so: check for the given issues in the general memory corruption section. Native code is a little harder to spot when compiled. If you have the sources then you can see that C files use .c source files and .h header files and C++ uses .cpp files and .h files. This is a little different from the .swift and the .m source files for Swift and Objective-C. These files can be part of the sources, or part of third party libraries, registered as frameworks and imported through various tools, such as Carthage, the Swift Package Manager or Cocoapods.
+
+Is Objective-C or Swift in use? Check for the following items:
+- The doubleFree issue: when `free()` is called twice for a given region instead of once.
+- Retaining cycles: look for cyclic dependencies by means of strong references of components to one another which keep materials in memory.
+- Using instances of `UnsafePointer` can be managed wrongly, which will allow for various memory corruption issues.
+- Trying to manage the reference count to an object by `Unmanaged` manually, leading to wrong counter numbers and a too late/too soon release.
+
+[A great talk is given on this subject at Realm academy](https://academy.realm.io/posts/russ-bishop-unsafe-swift/ "Russh Bishop on Unsafe Swift") and [a nice tutorial to see what is actually happening](https://www.raywenderlich.com/780-unsafe-swift-using-pointers-and-interacting-with-c "Unsafe Swift: Using Pointers And Interacting With C") is provided by Ray Wenderlich on this subject.
+
+>Please note that with Swift 5 you can only deallocate full blocks, which means the playground has changed a bit.
+
+#### Dynamic Analysis
+There are various tools provided which which memory bugs can be found:
+- Apple its Debug Memory graph introduced in XCode 8.
+- The Allocations and Leaks instrument in XCode.
```

Comment:
This is Instruments in Xcode, right? Then just the following should be sufficient:
- Instruments as part of Xcode